### ISSUE TITLE
Problem when opening a devcontainer on Ubuntu 20.04

### ISSUE THREAD

**Issue by andreriesco** [microsoft/vscode-remote-release#6814]
- VSCode Version: 1.68.0
- Local OS Version: Ubuntu 20.04 LTS
- Remote OS Version: Ubuntu 20.04 LTS
- Remote Extension/Connection Type: Containers
- Remote - Containers: 0.238.2
- Logs:
```
[158 ms] Remote-Containers 0.238.2 in VS Code 1.68.0 (4af164ea3a06f701fe3e89a2bcbb421d2026b68f).
[157 ms] Start: Resolving Remote
[235 ms] Setting up container for folder or workspace: /home/andreriesco/Documents/IDEExtension/test-applications/testqt
[244 ms] Start: Check Docker is running
[245 ms] Start: Run: docker version --format {{.Server.APIVersion}}
[333 ms] Server API version: 1.42
[336 ms] Start: Run: docker volume ls -q
[439 ms] Start: Run: docker ps -q -a --filter label=vsch.local.folder=/home/andreriesco/Documents/IDEExtension/test-applications/testqt --filter label=vsch.quality=stable
[527 ms] Start: Run: docker ps -q -a --filter label=devcontainer.local_folder=/home/andreriesco/Documents/IDEExtension/test-applications/testqt
[612 ms] Start: Run: /snap/code/98/usr/share/code/code /home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /home/andreriesco/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --workspace-folder /home/andreriesco/Documents/IDEExtension/test-applications/testqt --workspace-mount-consistency cached --id-label devcontainer.local_folder=/home/andreriesco/Documents/IDEExtension/test-applications/testqt --log-level debug --log-format json --config /home/andreriesco/Documents/IDEExtension/test-applications/testqt/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[840 ms] remote-containers 0.238.2.
[840 ms] Start: Run: docker buildx version
[985 ms] Start: Resolving Remote
[990 ms] Start: Run: git rev-parse --show-cdup
[1000 ms] Start: Run: docker ps -q -a --filter label=devcontainer.local_folder=/home/andreriesco/Documents/IDEExtension/test-applications/testqt
[1083 ms] Start: Run: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /home/andreriesco/Documents/IDEExtension/test-applications/testqt/.devcontainer/Dockerfile -t vsc-testqt-4a36f7b04b155e6106a2d9e2b756e526 /home/andreriesco/Documents/IDEExtension/test-applications/testqt/.devcontainer
[+] Building 0.1s (5/6)                                                         
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 872B                                       0.0s
 => [internal] load metadata for docker.io/library/testqt_arm32v7-qt5-way  0.0s
 => [1/2] FROM docker.io/library/testqt_arm32v7-qt5-wayland-no-ssh_bullse  0.0s
 => CACHED [2/2] RUN addgroup docker && groupmod --non-unique --gid 135 d  0.0s
 => preparing layers for inline cache                                      0.1s
error: failed to receive status: rpc error: code = Unavailable desc = error reading from server: EOF
[1424 ms] Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /home/andreriesco/Documents/IDEExtension/test-applications/testqt/.devcontainer/Dockerfile -t vsc-testqt-4a36f7b04b155e6106a2d9e2b756e526 /home/andreriesco/Documents/IDEExtension/test-applications/testqt/.devcontainer
[1425 ms]     at jR (/home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:240:1143)
[1425 ms]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[1425 ms]     at async Uu (/home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:239:2142)
[1426 ms]     at async Jy (/home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:239:1168)
[1426 ms]     at async zR (/home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:2006)
[1426 ms]     at async nb (/home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:3110)
[1427 ms]     at async JR (/home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:13046)
[1427 ms]     at async KR (/home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:12802)
[1435 ms] Exit code 1
[1445 ms] Command failed: /snap/code/98/usr/share/code/code /home/andreriesco/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /home/andreriesco/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --workspace-folder /home/andreriesco/Documents/IDEExtension/test-applications/testqt --workspace-mount-consistency cached --id-label devcontainer.local_folder=/home/andreriesco/Documents/IDEExtension/test-applications/testqt --log-level debug --log-format json --config /home/andreriesco/Documents/IDEExtension/test-applications/testqt/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[1446 ms] Exit code 1
```

Steps to Reproduce:

1. Try to open a project that contains a .devcontainer.json file on Visual Studio Code

Downgrading the Remote - Containers extension to v0.234.0 fixes the problem

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

**Comment by zoosewu** [comment]
same issue

**Comment by qbloomfield** [comment]
Also having this issue.  For me:
```bash
docker version
Client: Docker Engine - Community
 Version:           22.06.0-beta.0
 API version:       1.42
 Go version:        go1.18.3
 Git commit:        3e9117b
 Built:             Fri Jun  3 18:00:23 2022
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          22.06.0-beta.0
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.18.3
  Git commit:       38633e7
  Built:            Fri Jun  3 17:58:13 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

docker buildx version
github.com/docker/buildx v0.8.2 %{_buildx_gitcommit}
```

**Comment by chrmarti** [comment]
Could you append your devcontainer.json and Dockerfile? Also: If you have a non-beta install of Docker available, could you check if it works there?

**Comment by leograba** [comment]
Same issue, here are the versions:

```
docker version
Client: Docker Engine - Community
 Version:           20.10.16
 API version:       1.41
 Go version:        go1.17.10
 Git commit:        aa7e414
 Built:             Thu May 12 09:17:23 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.16
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.10
  Git commit:       f756502
  Built:            Thu May 12 09:15:28 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.4
  GitCommit:        212e8b6fa2f44b9c21b2798135fc6fb7c53efc16
 runc:
  Version:          1.1.1
  GitCommit:        v1.1.1-0-g52de29d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

docker buildx version
github.com/docker/buildx v0.3.0 c967f1d570fb393b4475c01efea58781608d093c
```

**Comment by chrmarti** [comment]
Could you append your devcontainer.json and Dockerfile?

**Comment by chrmarti** [comment]
Could you append your devcontainer.json and Dockerfile? @leograba @zoosewu Could you also include your log? (`F1` > `Remote-Containers: Show Container Log`)

**Comment by andreriesco** [comment]
devcontainer.json:
```
{
    "name": "torizon_testqt",
    "dockerFile": "Dockerfile",
    "extensions": [
        "ms-vscode.cpptools",
        "Toradex.torizon-early-access"
    ],
    "containerEnv": {
        "AR": "arm-linux-gnueabihf-ar",
        "AS": "arm-linux-gnueabihf-as",
        "CC": "arm-linux-gnueabihf-gcc",
        "CXX": "arm-linux-gnueabihf-g++",
        "CPP": "arm-linux-gnueabihf-cpp",
        "STRIP": "arm-linux-gnueabihf-strip",
        "LD": "arm-linux-gnueabihf-ld",
        "CROSS_COMPILE": "arm-linux-gnueabihf-",
        "QMAKE": "arm-linux-gnueabihf-qmake",
        "QT_INCLUDES": "/usr/include/arm-linux-gnueabihf/qt5"
    },
    "runArgs": [
        "--network=host"
    ],
    "remoteUser": "torizon",
    "mounts": [
        "source=${localEnv:HOME}${localEnv:USERPROFILE}/.moses,target=/home/torizon/.moses,type=bind",
        "source=//var/run/docker.sock,target=/var/run/docker.sock,type=bind"
    ],
    "postCreateCommand": "true"
}
```
Dockerfile (first one, that generates the image used on the FROM of the second one):
```
FROM torizon/debian-cross-toolchain-armhf:2-bullseye

RUN HOLD_PKGS='libdrm-common libdrm-amdgpu1:armhf libdrm2:armhf' \
    && apt-get -y update \
    && apt-get -y upgrade \
    && for P in $HOLD_PKGS ; do \
    echo ${P}=$(apt-cache show $P | sed -r -e '/^Version:/!d' -e 's/.* //' -e '/toradex/d' -e 'q') ; \
    done | xargs -r apt-get install -y --no-install-recommends \
    && apt-mark hold $HOLD_PKGS \
    && apt-get clean && apt-get autoremove && rm -rf /var/lib/apt/lists/*

# commands that should be run before installing packages (ex: to add a feed or keys)


# your regular RUN statements here
# Install required packages
RUN apt-get -q -y update \
    && apt-get -q -y install \
    python3-minimal \
    gdb-multiarch \
    procps \
    rsync \
    openssh-client \
    qt5-qmake:armhf \
    qtbase5-dev:armhf \
    && rm -rf /var/lib/apt/lists/*

RUN if [ ! -z "qtdeclarative5-dev:armhf libegl-dev:armhf libgl-dev:armhf" ]; then \
    apt-get -q -y update \
    && apt-get -q -y install qtdeclarative5-dev:armhf libegl-dev:armhf libgl-dev:armhf \
    && rm -rf /var/lib/apt/lists/* ; \
    fi

RUN if [ ! -z "" ]; then \
    apt-get -q -y update \
    && apt-get -q -y install  \
    && rm -rf /var/lib/apt/lists/* ; \
    fi

# add qt pretty printer
RUN mkdir -p /home/torizon/.gdb/qt5printers/ && \
    cd /home/torizon/.gdb/qt5printers && \
echo 'python \n\
import sys, os.path \n\
sys.path.insert(0, os.path.expanduser("~/.gdb")) \n\
import qt5printers \n\
qt5printers.register_printers(gdb.current_objfile()) \n\
end' > /home/torizon/.gdbinit

ADD sdkfiles /home/torizon/.gdb/qt5printers/
```
Dockerfile (second one):
```
FROM testqt_arm32v7-qt5-wayland-no-ssh_bullseye_release_abb31ba6-da93-4470-be15-8c9274fe10f8_sdk_image

# This is required to allow regular user inside the container to access docker socket
RUN addgroup docker && groupmod --non-unique --gid 135 docker && adduser torizon docker
```
docker version:
```
Client: Docker Engine - Community
 Version:           22.06.0-beta.0
 API version:       1.42
 Go version:        go1.18.3
 Git commit:        3e9117b
 Built:             Fri Jun  3 17:55:45 2022
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          22.06.0-beta.0
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.18.3
  Git commit:       38633e7
  Built:            Fri Jun  3 17:55:45 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
  ```
docker buildx version:
```
github.com/docker/buildx v0.8.2 
```

**Comment by chrmarti** [comment]
@andreriesco Could you try with a non-beta version of Docker?

**Comment by finickyDrone** [comment]
I'm seeing the exact same issue.
VSCode Version: 1.68.1
Local OS Version: Ubuntu 20.04 LTS
Remote OS Version: Ubuntu 20.04 LTS
Remote Extension/Connection Type: Containers
Remote - Containers: 0.238.2
Logs:
```
[2022-06-20T13:32:51.063Z] Remote-Containers 0.238.2 in VS Code 1.68.1 (30d9c6cd9483b2cc586687151bcbcd635f373630).
[2022-06-20T13:32:51.063Z] Start: Resolving Remote
[2022-06-20T13:32:51.067Z] Setting up container for folder or workspace: /home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus
[2022-06-20T13:32:51.069Z] Start: Check Docker is running
[2022-06-20T13:32:51.069Z] Start: Run: docker version --format {{.Server.APIVersion}}
[2022-06-20T13:32:51.092Z] Stop (23 ms): Run: docker version --format {{.Server.APIVersion}}
[2022-06-20T13:32:51.093Z] Server API version: 1.41
[2022-06-20T13:32:51.093Z] Stop (24 ms): Check Docker is running
[2022-06-20T13:32:51.093Z] Start: Run: docker volume ls -q
[2022-06-20T13:32:51.111Z] Stop (18 ms): Run: docker volume ls -q
[2022-06-20T13:32:51.114Z] Start: Run: docker ps -q -a --filter label=vsch.local.folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --filter label=vsch.quality=stable
[2022-06-20T13:32:51.131Z] Stop (17 ms): Run: docker ps -q -a --filter label=vsch.local.folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --filter label=vsch.quality=stable
[2022-06-20T13:32:51.132Z] Start: Run: docker ps -q -a --filter label=devcontainer.local_folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus
[2022-06-20T13:32:51.150Z] Stop (18 ms): Run: docker ps -q -a --filter label=devcontainer.local_folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus
[2022-06-20T13:32:51.150Z] Start: Run: docker inspect --type container 3bd774d820e8
[2022-06-20T13:32:51.168Z] Stop (18 ms): Run: docker inspect --type container 3bd774d820e8
[2022-06-20T13:32:51.169Z] Start: Run: /usr/share/code/code /home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /home/finickyDrone/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --workspace-folder /home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --workspace-mount-consistency cached --id-label devcontainer.local_folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --log-level debug --log-format json --config /home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[2022-06-20T13:32:51.265Z] remote-containers 0.238.2.
[2022-06-20T13:32:51.265Z] Start: Run: docker buildx version
[2022-06-20T13:32:51.311Z] Stop (46 ms): Run: docker buildx version
[2022-06-20T13:32:51.311Z] Start: Resolving Remote
[2022-06-20T13:32:51.313Z] Start: Run: git rev-parse --show-cdup
[2022-06-20T13:32:51.315Z] Stop (2 ms): Run: git rev-parse --show-cdup
[2022-06-20T13:32:51.316Z] Start: Run: docker ps -q -a --filter label=devcontainer.local_folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus
[2022-06-20T13:32:51.333Z] Stop (17 ms): Run: docker ps -q -a --filter label=devcontainer.local_folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus
[2022-06-20T13:32:51.333Z] Start: Run: docker inspect --type container 3bd774d820e8
[2022-06-20T13:32:51.351Z] Stop (18 ms): Run: docker inspect --type container 3bd774d820e8
[2022-06-20T13:32:51.351Z] Start: Starting container
[2022-06-20T13:32:51.352Z] Start: Run: docker start 3bd774d820e85aa33035686e7674290646544892d4a33728f772bea0aa6aaa26
[2022-06-20T13:32:51.381Z] Stop (29 ms): Run: docker start 3bd774d820e85aa33035686e7674290646544892d4a33728f772bea0aa6aaa26
[2022-06-20T13:32:51.382Z] Error: Command failed: docker start 3bd774d820e85aa33035686e7674290646544892d4a33728f772bea0aa6aaa26
[2022-06-20T13:32:51.382Z]     at NR (/home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:239:1572)
[2022-06-20T13:32:51.382Z]     at Jy (/home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:239:1510)
[2022-06-20T13:32:51.382Z]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2022-06-20T13:32:51.382Z]     at async zR (/home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:2006)
[2022-06-20T13:32:51.382Z]     at async nb (/home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:3110)
[2022-06-20T13:32:51.382Z]     at async JR (/home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:13046)
[2022-06-20T13:32:51.382Z]     at async KR (/home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js:245:12802)
[2022-06-20T13:32:51.385Z] Stop (216 ms): Run: /usr/share/code/code /home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /home/finickyDrone/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --workspace-folder /home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --workspace-mount-consistency cached --id-label devcontainer.local_folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --log-level debug --log-format json --config /home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[2022-06-20T13:32:51.385Z] Exit code 1
[2022-06-20T13:32:51.386Z] Command failed: /usr/share/code/code /home/finickyDrone/.vscode/extensions/ms-vscode-remote.remote-containers-0.238.2/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /home/finickyDrone/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --workspace-folder /home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --workspace-mount-consistency cached --id-label devcontainer.local_folder=/home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus --log-level debug --log-format json --config /home/finickyDrone/Documents/LocalProjects/rokus/Software/rokus/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[2022-06-20T13:32:51.387Z] Exit code 1

```

Downgrading back to v0.234.0 makes things work again, so I think v0.238.2 introduced an error.

**Comment by Zincr0** [comment]
switching docker to 20.10.17 did nothing here for me, only downgrading "Remote - Containers" extension to v0.234.0 "fixed" the problem

**Comment by Tielem** [comment]
Also affected by this bug.

I run Docker remotely, and create my dev containers with the command pallet, Remote-Containers: Clone repository in container volume. I saw similar error messages as finickyDrone when doing so.

Downgrading the Remote - Containers extension to v0.234.0 resolved the issue for me, as suggested by Zincr0.

**Comment by chrmarti** [comment]
Maybe related: https://github.com/moby/moby/pull/42951

**Comment by chrmarti** [comment]
What do you get for `docker buildx ls`? (Could be you are using a builder that is isolated from the local image cache.)

**Comment by Didjacome** [comment]
I was able to solve my case.

I saw that the project folder was created with %20 due to git colone.
I created a new folder called project and git cloned it.
I copied the .devcontainer folder to project and it worked here.

I was left with the folder structure like this:
![image](https://user-images.githubusercontent.com/83463639/197355638-f6d3154f-9f8e-4d39-a578-d3fd7eca7efc.png)


docker: 20.10.17
vscode: 1.72.2
dev container extension: 234.0
operating system: Microsoft Windows 10 Enterprise
version OS: 10.0.19044 N/A build 19044
wsl version: 2
WSL operating system: Ubuntu 22.04.1 LTS release.

**Comment by waynejhou** [comment]
Somehow it this problem happened on me after install lastest vscode and dev container extenstion.
After some guessing, I solved the problem by disable `BUILDKIT_INLINE_CACHE` in docker buildx
I don't really known what this argument can do, but it fix problem.
```json
"build": {
    "context": "./path/to/context",
    "dockerfile": "./path/to/dockerfile",
    "args": {
        "BUILDKIT_INLINE_CACHE": "0"
    }
}
```
docker: `23.0.0, build e92dd87`
buildx version: `github.com/docker/buildx v0.10.2 00ed17d`
vscode: `1.75.0`
dev container extension: `0.275.0`
operating system: `Ubuntu 22.04.1 LTS x86_64`
kernal: `5.15.0-58-generic`

**Comment by mlesin** [comment]
> Somehow it this problem happened on me after install lastest vscode and dev container extenstion. After some guessing, I solved the problem by disable `BUILDKIT_INLINE_CACHE` in docker buildx I don't really known what this argument can do, but it fix problem.

Same problem here, with exactly the same versions of docker, buildx, vscode, extension and OS with kernel :)
But in my case it's even worse: I'm using a docker-compose based build environment, and vscode always adds a `docker-compose.devcontainer.build-[build-id].yml` file with overrides, where it says the following::
```docker-compose.yml
version: "3.8"

services:
  app:
    build:
      dockerfile: /tmp/devcontainercli-mike/container-features/0.29.0-1675421070034/Dockerfile-with-features
      args:
        - BUILDKIT_INLINE_CACHE=1
        - _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label

```
I have tried to comment BUILDKIT_INLINE_CACHE string, then I can build image manually, but the problem is vscode always rewrites it during startup, and I have no idea how to disable BUILDKIT_INLINE_CACHE with docker-compose based devcontainer :(

**Comment by eiswind** [comment]
Same problem here. Are any workarounds known?

**Comment by tobiasrdm** [comment]
A workarround is disable docker buildkit before open vscode. Example:
`DOCKER_BUILDKIT=0 code .`

**Comment by mrjackwills** [comment]
> A workarround is disable docker buildkit before open vscode. Example: `DOCKER_BUILDKIT=0 code .`

You can also put this in your devcontainter.json,

```json

"build": {
    "dockerfile": "Dockerfile",
    "args": {
         "VARIANT": "bullseye",
	 "BUILDKIT_INLINE_CACHE": "0"
    }
},

```

**Comment by darioleanbit** [comment]
> > A workarround is disable docker buildkit before open vscode. Example: `DOCKER_BUILDKIT=0 code .`
> 
> You can also put this in your `devcontainer.json`,
> 
> ```json
> "build": {
>     "dockerfile": "Dockerfile",
>     "args": {
>          "VARIANT": "bullseye",
> 	 "BUILDKIT_INLINE_CACHE": "0"
>     }
> },
> ```
> 
> Fixes the issue, although not 100% sure what it actually does

Same thing here, adding `"BUILDKIT_INLINE_CACHE": "0"` seems to fix the issue.

**Note:** in my case, when the error happened, docker killed also other container that were running on my system (not very good :sweat:)... maybe a Docker bug?

**Comment by tt4g** [comment]
> Note: in my case, when the error happened, Docker killed also other containers that were running on my system (not very good 😓)... maybe a Docker bug?

@darioleanbit Yes. The underlying error is caused by Docker.
The buildx plugin added in Docker v23 is causing problems.
I got to the PR moby/moby#44920 that fixes the problem from [this comment](https://github.com/microsoft/vscode-remote-release/issues/7958#issuecomment-1416495001).

**Comment by chrmarti** [comment]
This should be fixed in Docker v23.0.1. (https://github.com/microsoft/vscode-remote-release/issues/7958#issuecomment-1425585936) Closing, let me know if you still see this with that version of Docker.
